browser-test/src/applicant_application_index.test.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  createTestContext,
+  loginAsAdmin,
+  loginAsGuest,
+  loginAsTestUser,
+  logout,
+  selectApplicantLanguage,
+} from './support'
+
+describe('applicant program index page', () => {
+  const ctx = createTestContext()
+
+  const primaryProgramName = 'application-index-primary-program'
+  const otherProgramName = 'application-index-other-program'
+
+  beforeAll(async () => {
+    const {page, adminPrograms, adminQuestions} = ctx
+    await loginAsAdmin(page)
+
+    // Create a program with two questions on separate blocks so that an applicant can partially
+    // complete an application.
+    await adminPrograms.addProgram(primaryProgramName)
+    await adminQuestions.addTextQuestion({questionName: 'first-q'})
+    await adminQuestions.addTextQuestion({questionName: 'second-q'})
+    await adminPrograms.addProgramBlock(primaryProgramName, 'first block', [
+      'first-q',
+    ])
+    await adminPrograms.addProgramBlock(primaryProgramName, 'second block', [
+      'second-q',
+    ])
+
+    await adminPrograms.addProgram(otherProgramName)
+
+    await adminPrograms.gotoAdminProgramsPage()
+    await adminPrograms.publishAllPrograms()
+    await logout(page)
+  })
+
+  it('categorizes programs for draft and applied applications', async () => {
+    const {page, applicantQuestions} = ctx
+    await loginAsTestUser(page)
+    await selectApplicantLanguage(
+      page,
+      'English',
+      /* assertProgramIndexPage= */ true,
+    )
+
+    // Navigate to the applicant's program index and validate that both programs appear in the
+    // "Not started" section.
+    await applicantQuestions.expectPrograms({
+      notStartedPrograms: [primaryProgramName, otherProgramName],
+      inProgressPrograms: [],
+      submittedPrograms: [],
+    })
+
+    // Fill out first application block and confirm that the program appears in the "In progress"
+    // section.
+    await applicantQuestions.applyProgram(primaryProgramName)
+    await applicantQuestions.answerTextQuestion('first answer')
+    await applicantQuestions.clickNext()
+    await applicantQuestions.gotoApplicantHomePage()
+    await applicantQuestions.expectPrograms({
+      notStartedPrograms: [otherProgramName],
+      inProgressPrograms: [primaryProgramName],
+      submittedPrograms: [],
+    })
+
+    // Finish the application and confirm that the program appears in the "Submitted" section.
+    await applicantQuestions.applyProgram(primaryProgramName)
+    await applicantQuestions.answerTextQuestion('second answer')
+    await applicantQuestions.clickNext()
+    await applicantQuestions.submitFromReviewPage()
+    await applicantQuestions.returnToProgramsFromSubmissionPage()
+    await applicantQuestions.expectPrograms({
+      notStartedPrograms: [otherProgramName],
+      inProgressPrograms: [],
+      submittedPrograms: [primaryProgramName],
+    })
+
+    // Logout, then login as guest and confirm that everything appears unsubmitted. This is a
```

Comment:
I'd remove the "regression" part.  It's a completely valid and good test regardless. The regression only makes sense during this review, and likely isn't helpful after.

Reply:
Totally agree that this is a valid and good test. That said, I also want to specifically call out that it's preventing us from backsliding on a potentially bad issue for context. I modified it to take out the word "regression" and put the issue in a parenthetical.

Reply:
That statement applies to most of our tests though which makes it noisy.    It's why we have browser tests for the most part :)

IMO making the intent of the test clear is how we address this concern. I'd say this is a great reason to have focused tests even if setup is redundant; and it makes it readable going forward.  In another year of development, no one is going to have context on this regression versus the tests of all the new features.

up to you though.

Reply:
K, I think I'm going to keep this one since I think the severity of a regression warrants easily finding the context for why it was added in the first place.